tests/api-diff/arrow_tests.t

```diff
@@ -0,0 +1,47 @@
+Here we generate a `.mli` file with an arrow type:
```

Comment:
It might be best to merge the type expression diffing tests together under `fine_grained_type_expr.t` so we have a natural place to test how they all tie together, e.g. whether a tuple within an arrow is well displayed or the otherway around.